src/main/kotlin/no/nav/familie/ks/sak/kjerne/brev/BrevPeriodeContext.kt

```diff
@@ -276,20 +278,34 @@ class BrevPeriodeContext(
                 val nullableSanitybegrunnelse = vedtakBegrunnelse.begrunnelse.tilSanityBegrunnelse(sanityBegrunnelser)
                 nullableSanitybegrunnelse?.let { Pair(vedtakBegrunnelse.begrunnelse, it) }
             }.map { (begrunnelse, sanityBegrunnelse) ->
-                val personerMedVilkårSomPasserBegrunnelse = begrunnelserForPeriodeContext
-                    .hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(
+
+                val relevantePersoner = when (begrunnelse.begrunnelseType) {
+                    BegrunnelseType.ETTER_ENDRET_UTBETALING -> begrunnelserForPeriodeContext.hentPersonerMedEndretUtbetalingerSomPasserMedVedtaksperiode(
+                        sanityBegrunnelse
+                    )
+
+                    else -> begrunnelserForPeriodeContext.hentPersonerMedVilkårResultaterSomPasserMedBegrunnelseOgPeriode(
                         begrunnelse = begrunnelse,
                         sanityBegrunnelse = sanityBegrunnelse
                     )
+                }
 
                 val antallTimerBarnehageplass =
-                    hentAntallTimerBarnehageplassTekst(personerMedVilkårSomPasserBegrunnelse)
+                    hentAntallTimerBarnehageplassTekst(relevantePersoner)
+
+                val endringsperioder = begrunnelse.hentRelevanteEndringsperioderForBegrunnelse(
+                    endretUtbetalingAndeler = andelTilkjentYtelserMedEndreteUtbetalinger.flatMap { it.endreteUtbetalinger },
+                    vedtaksperiode = utvidetVedtaksperiodeMedBegrunnelser
+                )
 
-                val gjelderSøker = personerMedVilkårSomPasserBegrunnelse.any { it.type == PersonType.SØKER }
+                val søknadstidspunkt = endringsperioder.sortedBy { it.søknadstidspunkt }
+                    .firstOrNull { sanityBegrunnelse.endringsårsaker.contains(it.årsak) }?.søknadstidspunkt
```

Comment:
Burde vi ikke filtrere på `sanityBegrunnelse.endringsårsaker.contains(it.årsak)` før vi får endringsperioder? 🤔 

I så fall kan man bruke `minBy{it.søknadstidspunkt}` om filtreringen gjøres tidligere